cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiff.java

```diff
@@ -0,0 +1,255 @@
+                        // Some default properties do not have set ConfigEntry.ConfigSource.DEFAULT_CONFIG and thus
+                        // we are removing property. That might cause redundant RU. To fix this we would have to add defaultValue
+                        // to the configModel
+                        log.trace("{} not set in desired, using default value", entry.name());
+                    } else {
+                        // entry is in current, is not in desired, is not default -> it was using non-default value and was removed
+                        // if the entry was custom, it should be deleted
+                        if (!isIgnorableProperty(pathValueWithoutSlash)) {
+                            updatedCE.add(new AlterConfigOp(new ConfigEntry(pathValueWithoutSlash, null), AlterConfigOp.OpType.DELETE));
+                            log.trace("{} not set in desired, unsetting back to default {}", entry.name(), "deleted entry");
+                        } else {
+                            log.trace("{} is ignorable, not considering as removed");
+                        }
+                    }
+                } else if (d.get("op").asText().equals("replace")) {
+                    // entry is in the current, desired is updated value
+                    if (!isIgnorableProperty(pathValueWithoutSlash)) {
+                        if (isEntryCustom(entry)) {
+                            log.trace("custom property {} has new desired value {}", entry.name(), desiredMap.get(entry.name()));
+                        } else {
+                            log.trace("property {} has new desired value {}", entry.name(), desiredMap.get(entry.name()));
+                            updatedCE.add(new AlterConfigOp(new ConfigEntry(pathValueWithoutSlash, desiredMap.get(pathValueWithoutSlash)), AlterConfigOp.OpType.SET));
+                        }
+                    } else {
+                        log.trace("{} is ignorable, not considering as replaced");
+                    }
+                }
+            } else {
+                if (d.get("op").asText().equals("add")) {
+                    // entry is not in the current, it is added
+                    if (!isIgnorableProperty(pathValueWithoutSlash)) {
+                        if (isEntryCustom(pathValueWithoutSlash)) {
+                            log.trace("add new custom property {} {}", pathValueWithoutSlash, d.get("op").asText());
+                        } else {
+                            log.trace("add new {} {}", pathValueWithoutSlash, d.get("op").asText());
+                            updatedCE.add(new AlterConfigOp(new ConfigEntry(pathValueWithoutSlash, desiredMap.get(pathValueWithoutSlash)), AlterConfigOp.OpType.SET));
+                        }
+                    } else {
+                        log.trace("{} is ignorable, not considering as added");
+                    }
```

Comment:
This is almost identical to the `d.get("op").asText().equals("replace")` case, you can factor out a single method to encapsulate both.